src/xds/client.rs

```diff
@@ -266,6 +265,26 @@ impl AdsClient {
                 tokio::time::sleep(backoff).await;
                 backoff
             }
+            Err(e @ Error::GrpcStatus(_)) => {
+                let err_detail = e.to_string();
+                let status: tonic::Status = e.try_into().unwrap();
+                if status.code() == tonic::Code::Unknown
```

Comment:
why Unknown? we don't have that in the Go version and it seems important

Reply:
https://github.com/istio/ztunnel/issues/400  unknown error here, not sure the error in  rust is same as go 

```
2023-02-15T20:54:40.204428Z WARN xds{id=1}: ztunnel::xds::client: XDS client error: gRPC error (Unknown error): error reading a body from connection: broken pipe, retrying```